Firebase/Auth/Source/FIRAuthOperation.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/** @class FIRAuthOperation
+    @brief A class which provides operation types for RPCs that support the operation parameter.
+ */
+@interface FIRAuthOperation : NSObject
```

Comment:
You don't need a class for declaring an enum, which is global anyway.

And you can call the file FIRAuthOperationType.h if you don't have the FIRAuthOperation class.